File: python/pysearch/searchfile.py
################################################################################
#
# searchfile.py
#
# class SearchFile: encapsulates a file to search
#
################################################################################
from cStringIO import StringIO
import os

from fileutil import FileType, FileUtil

class SearchFile:
    """encapsulates a search file"""
    CONTAINER_SEPARATOR = '!'

    def __init__(self, **kargs):
        self.containers = []
        self.path = ''
        self.filename = ''
        self.filetype = FileType.Unknown
        self.__dict__.update(kargs)

    @property
    def relativepath(self):
        return os.path.join(self.path, self.filename)

    def __str__(self):
        sio = StringIO()
        if self.containers:
            sio.write(self.CONTAINER_SEPARATOR.join(self.containers))
            sio.write(self.CONTAINER_SEPARATOR)
        sio.write(self.relativepath)
        return sio.getvalue()

File: php/phpsearch/searchfile.php
<?php

/**
 * Class SearchFile
 *
 * @property array containers
 * @property string path
 * @property string filename
 * @property filetype
 */
class SearchFile
{
    const CONTAINER_SEPARATOR = '!';

    public function __construct(string $path, string $filename, $filetype)
    {
        $this->containers = array();
        $this->path = $path;
        $this->filename = $filename;
        $this->filetype = $filetype;
    }

    public function __toString(): string
    {
        $s = "";
        if ($this->containers) {
            $s = join(CONTAINER_SEPARATOR, $this->containers) . CONTAINER_SEPARATOR;
        }
        $s .= FileUtil::join_path($this->path, $this->filename);
        return $s;
    }
}
